src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```diff
@@ -213,10 +227,7 @@ else if (ast.getType() == JavadocTokenTypes.SINCE_LITERAL) {
     public void finishJavadocTree(DetailNode rootAst) {
         moduleDetails.setDescription(getDescriptionText());
         if (isTopLevelClassJavadoc()) {
-            if (getFileContents().getFileName().contains("test")) {
```

Comment:
This is the main reason the property is needed to drop having this code that checks the file name which will cause issues with checks with that word in them. New property always defaults to `write`.